### PR TITLE
Add JavaScript example for modals

### PR DESCRIPTION
**Pull request by luap42**
Currently, there's no way to see, how modals work outside of the example box. (See: https://design.codidact.org/components.modals.html)

This PR adds a "launch X modal button" below every example:

grafik

This opens a matching modal:

grafik

The included JavaScript is example-only and not intended for production.

### CODE REVIEW

File: docs/assets/js/docs.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    modal = document.querySelector(this.getAttribute("data-trigger-modal"));
+
+    if(!modal.classList.contains("is-with-backdrop"))
+      modal.style.top = document.scrollingElement.scrollTop + "px";
+
+    modal.classList.toggle("is-active");
+
+    e.preventDefault();
+  })
+}
```
Comment (ArtOfCode-) [inline docs/assets/js/docs.js]: This whole section needs revision to match up to the JS standards. Specifically:

 * Declare variables as `const`, not `var` (L3) or nothing (L1)
 * Spaces: after `for`, around `=`, after `function`
 * Variable names are `camelCase` not `snake_case`
 * Missing braces around single-line `if` statement body
 * Missing semicolon after the `addEventListener` call
 * Recommend using descriptive names (what are `m_triggers` and `mt`?)

Comment (luap42) [reply docs/assets/js/docs.js]: Oh yeah, sorry. Will do later today/tomorrow. I think the same applies to the JS in #6, so no need to review-critizise it there, until fixed here.

Comment (luap42) [reply docs/assets/js/docs.js]: Should be fixed now. Please re-review. :)

File: docs/assets/js/docs.js [hunk 2]
```diff
@@ -1,16 +1,22 @@
-m_triggers = document.querySelectorAll("[data-trigger-modal]");
+/* Modals */
+const modalTriggers = document.querySelectorAll("[data-trigger-modal]");
 
-for(var i=0; i < m_triggers.length; i++) {
-  mt = m_triggers[i];
+for (const i = 0; i < modalTriggers.length; i++) {
```
Comment (ArtOfCode-) [inline docs/assets/js/docs.js]: This one needs to be a `let`, because it's reassigned each iteration of the loop.

File: docs/assets/js/docs.js [hunk 3]
```diff
@@ -1,16 +1,22 @@
...
+  modalTrigger = modalTriggers[i];
 
-  mt.addEventListener("click", function(e) {
+  modalTrigger.addEventListener("click", function (e) {
```
Comment (ArtOfCode-) [inline docs/assets/js/docs.js]: Missing `const` on both of these assignments.

File: docs/assets/js/docs.js [hunk 4]
```diff
@@ -1,16 +1,22 @@
...
     modal = document.querySelector(this.getAttribute("data-trigger-modal"));
 
-    if(!modal.classList.contains("is-with-backdrop"))
+    // Set appropriate on-screen position for modals,
+    // which do not have .is-with-backdrop, and are
+    // therefore position absolute not fixed.
+    if(!modal.classList.contains("is-with-backdrop")) {
       modal.style.top = document.scrollingElement.scrollTop + "px";
```
Comment (ArtOfCode-) [inline docs/assets/js/docs.js]: No changes necessary here, but we'll want to consider for the JS standards whether we want to allow implicit type coercion like this or whether we require an explicit `toString()` or template literal use.

File: docs/assets/js/docs.js (hunk 3 above)
Comment (ArtOfCode-) [reply docs/assets/js/docs.js]: Ignore this - GH showed me the wrong diff for L5-7.

File: docs/assets/js/docs.js (hunk 2 above)
Comment (luap42) [reply docs/assets/js/docs.js]: Fixed

File: docs/assets/js/docs.js (hunk 3 above)
Comment (luap42) [reply docs/assets/js/docs.js]: Fixed, by removing the additional variable by direct access

File: docs/assets/js/docs.js (hunk 4 above)
Comment (luap42) [reply docs/assets/js/docs.js]: Gotcha.

**Pull request merged**